### PR TITLE
AmazonLinux Improvements

### PR DESCRIPTION
**Pull request by JakubKotkowiak**
- Support for amazonLinux2
- Maintain support for amazonLinux1

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -37,6 +37,10 @@
     {
       "operatingsystem": "Windows",
       "operatingsystemrelease": ["2008", "2012", "2016"]
+    },
+    {
+      "operatingsystem": "Amazon",
+      "operatingsystemrelease": ["1", "2"]
```
Comment (vchepkov) [inline metadata.json:43]: actually, it's not 1, it's 2018

Comment (JakubKotkowiak) [reply metadata.json:43]: Thank you, I forgot to resolve it in this file

File: manifests/agent.pp [hunk 2]
```diff
@@ -125,8 +125,10 @@
               }
             }
             'RedHat', 'CentOS', 'Amazon', 'OracleLinux': {
-              if ($::operatingsystem == 'Amazon') {
+              if ($::operatingsystem == 'Amazon' and $::operatingsystemmajrelease != '2'){
```
Comment (vchepkov) [inline manifests/agent.pp:128]: this will include future releases, wouldn't it?

Comment (JakubKotkowiak) [reply manifests/agent.pp:128]: Yes, I can change it to = '2018' but in that case it won't include the earlier ones

Comment (vchepkov) [reply manifests/agent.pp:128]: I don't have a good answer here, but metadata.json and AWS itself only claim support for versions 2018 and 2

**Review by varas — APPROVED** [review]

**Pull request merged**